### PR TITLE
Several referral program improvements and updates

### PR DESCRIPTION
**Pull request by tiagonmas**
Improved management of campaigns, and hand over of referral codes.

### CODE REVIEW

File: BancoAlimentar.AlimentaEstaIdeia.Web/Areas/Admin/Pages/Donations/Edit.cshtml [hunk 1]
```diff
@@ -29,11 +29,11 @@
                 <input asp-for="Donation.WantsReceipt" class="form-control" />
                 <span asp-validation-for="Donation.WantsReceipt" class="text-danger"></span>
             </div>
-            <div class="form-group">
+@*            <div class="form-group">
```
Comment (guerrerotook) [inline BancoAlimentar.AlimentaEstaIdeia.Web/Areas/Admin/Pages/Donations/Edit.cshtml:32]: If you don't need this section, just delete it.

File: BancoAlimentar.AlimentaEstaIdeia.Web/Pages/CashDonation.cshtml.cs [hunk 2]
```diff
@@ -373,7 +373,6 @@ public async Task<ActionResult> OnPost()
                         DonationDate = DateTime.UtcNow,
                         DonationAmount = Amount,
                         FoodBank = this.context.FoodBank.GetById(FoodBankId),
-                        Referral = referral_code,
```
Comment (guerrerotook) [inline BancoAlimentar.AlimentaEstaIdeia.Web/Pages/CashDonation.cshtml.cs:376]: referral_code is not valid in C#, it should be referralCode instead

File: BancoAlimentar.AlimentaEstaIdeia.Web/Pages/Referral.cshtml.cs [hunk 3]
```diff
@@ -25,16 +39,17 @@ public ActionResult OnGet(string text)
...
+                // Check if referral code exists. Give error if it does not. Redirect if it does exist.
+                var referral = this.context.ReferralRepository.GetActiveCampaignsByCode(text);
+                if (referral != null)
+                {
+                    this.HttpContext.Session.SetString("Referral", text);
+                    return this.RedirectToPage("./Donation");
+                }
+                else
+                {
+                    return null;
```
Comment (guerrerotook) [inline BancoAlimentar.AlimentaEstaIdeia.Web/Pages/Referral.cshtml.cs:51]: In here you return the page to render or the page to redirect. If the referral code doesn't exits then you should redirect to Index (Home Page instead).

File: BancoAlimentar.AlimentaEstaIdeia.Web/Areas/Admin/Pages/Donations/Edit.cshtml (hunk 1 above)
Comment (tiagonmas) [reply BancoAlimentar.AlimentaEstaIdeia.Web/Areas/Admin/Pages/Donations/Edit.cshtml]: removed

File: BancoAlimentar.AlimentaEstaIdeia.Web/Pages/Referral.cshtml.cs (hunk 3 above)
Comment (tiagonmas) [reply BancoAlimentar.AlimentaEstaIdeia.Web/Pages/Referral.cshtml.cs]: redirecting now to Index.

**Review by bancoalimentar — APPROVED** [review]

File: BancoAlimentar.AlimentaEstaIdeia.Web/Pages/Referral.cshtml.cs (hunk 3 above)
Comment (tiagonmas) [reply BancoAlimentar.AlimentaEstaIdeia.Web/Pages/Referral.cshtml.cs]: Actually, i did want to stay on the same page in this case, and show a message that the campaign is no longer active. I had that logic in the UI part. If I want to stay on the same page is returning null not the right thing ? @guerrerotook

Comment (tiagonmas) [reply BancoAlimentar.AlimentaEstaIdeia.Web/Pages/Referral.cshtml.cs]: I'll update to return Page();

**Pull request merged**